solang-parser/src/lexer.rs

```diff
@@ -796,20 +807,22 @@ impl<'input> Lexer<'input> {
                                             }
                                         }
 
-                                        return Some(Err(
+                                        self.errors.push(
                                             LexicalError::InvalidCharacterInHexLiteral(
                                                 Loc::File(self.file_no, i, i + 1),
                                                 ch,
                                             ),
-                                        ));
+                                        );
+                                       break;
```

Comment:
why break? I think you need to `continue` with a label

If I compile this I get:
```solidity
hex"g" 
```
I get:
```
error: end of file found in string literal
  ┌─ /home/sean/solang/overflow/foo.sol:1:1
  │  
1 │ ╭ hex"g" 
2 │ │ 
  │ ╰^

error: invalid character 'g' in hex literal string
  ┌─ /home/sean/solang/overflow/foo.sol:1:5
  │
1 │ hex"g" 
  │     ^

```

Test case for this please